src/bm_sim/P4Objects.cpp

```diff
@@ -2211,9 +2212,20 @@ P4Objects::check_required_fields(
 
 std::unique_ptr<CalculationsMap::MyC>
 P4Objects::check_hash(const std::string &name) const {
-  auto h = CalculationsMap::get_instance()->get_copy(name);
-  if (!h) throw json_exception(EFormat() << "Unknown hash algorithm: " << name);
-  return h;
+  {
+    auto h = CalculationsMap::get_instance()->get_copy(name);
+    if (h) return h;
+  }
+  {
+    auto h = CrcMap::get_instance()->get_copy(name);
+    if (h) return h;
+  }
+  {
+    auto h = CrcMap::get_instance()->get_copy_from_custom_str(name);
```

Comment:
Since you're not doing any additional checking before calling the name, is there a particular reason that you expose both get_copy and get_copy_form_custom_str? That is vs. get_copy checking the map and if there is a miss, directly calling get_copy_from_custom_str as a private method?